addons/mail/models/mail_template.py

```diff
@@ -64,6 +66,7 @@ def default_get(self, fields):
     auto_delete = fields.Boolean(
         'Auto Delete', default=True,
         help="This option permanently removes any track of email after it's been sent, including from the Technical menu in the Settings, in order to preserve storage space of your Odoo database.")
+    description = fields.Text('Template description', translate=True, help="This field is used for internal description of the template's usage.")
```

Comment:
Just to keep fields ordered, I would define it under "name" as it is part of the template definition